classes/Product.php

```diff
@@ -5549,17 +5549,6 @@ public static function getProductProperties($id_lang, $row, Context $context = n
             $cache_key .= '-pack' . $row['id_product_pack'];
         }
 
-        if (!isset($row['cover_image_id'])) {
-            $cover = static::getCover($row['id_product']);
-            if (isset($cover['id_image'])) {
-                $row['cover_image_id'] = $cover['id_image'];
-            }
-        }
-
-        if (isset($row['cover_image_id'])) {
-            $cache_key .= '-cover' . (int) $row['cover_image_id'];
-        }
-
```

Comment:
I'm not sure about the impact of removing this, just make sure that [this error](https://github.com/PrestaShop/PrestaShop/issues/21389) isn't coming back.

Reply:
It works fine.
If the template wants the cover -> $product.cover
If it wants a combination image -> $product.default_image